src/events/rewards/rewards.service.ts

```diff
@@ -90,41 +86,57 @@ export class RewardsService {
       };
     }
 
-    const { blockNumber, frames } = framesFromLastReport;
+    const rewards = await Promise.all(
+      framesFromLastReports.map(async ({ blockNumber, frames }) => {
+        const { clRewards, elRewards } = await this.getRewardsByBlockNumber(blockNumber, frames);
 
-    if (frames.eq(0)) {
-      this.logger.warn('last rewards set to 0 because frames passed from last event is 0.', {
-        service: RewardsService.SERVICE_LOG_NAME,
-      });
-      return {
-        clRewards: BigNumber.from(0),
-        elRewards: BigNumber.from(0),
-        allRewards: BigNumber.from(0),
-      };
-    }
+        return {
+          clRewards: clRewards,
+          elRewards: elRewards,
+        };
+      }),
+    );
 
-    const { preCLBalance, postCLBalance } = await this.getEthDistributed(blockNumber);
-    const elRewards = (await this.getElRewards(blockNumber)) ?? BigNumber.from(0);
-    const withdrawalsReceived = (await this.getWithdrawalsReceived(blockNumber)) ?? BigNumber.from(0);
+    let minCL = rewards[0].clRewards;
+    let minEL = rewards[0].elRewards;
 
-    const clValidatorsBalanceDiff = postCLBalance.sub(preCLBalance);
-    const clRewards = clValidatorsBalanceDiff.add(withdrawalsReceived);
+    rewards.forEach((r) => {
```

Comment:
pls add a comment about logic - why do you iterate by all rewards?
like - to find minimal for last week